AutoCollection/StatsBeat.ts

```diff
@@ -0,0 +1,218 @@
+    }
+
+    public enable(isEnabled: boolean) {
+        this._isEnabled = isEnabled;
+        if (this._isEnabled && !this._isInitialized) {
+            this._getCustomProperties();
+            this._isInitialized = true;
+        }
+        if (isEnabled) {
+            if (!this._handle) {
+                this._handle = setInterval(() => {
+                    this.trackStatsBeatMetrics()
+                }, this._collectionInterval);
+                this._handle.unref(); // Allow the app to terminate even while this loop is going on
+            }
+        } else {
+            if (this._handle) {
+                clearInterval(this._handle);
+                this._handle = null;
+            }
+        }
+    }
+
+    public isInitialized() {
+        return this._isInitialized;
+    }
+
+    public isEnabled() {
+        return this._isEnabled;
+    }
+
+    public setCodelessAttach() {
+        this._attach = Constants.StatsBeatAttach.codeless;
+    }
+
+    public addFeature(feature: Constants.StatsBeatFeature) {
+        this._features |= feature;
+    }
+
+    public removeFeature(feature: Constants.StatsBeatFeature) {
+        this._features &= ~feature;
+    }
+
+    public addInstrumentation(instrumentation: Constants.StatsBeatInstrumentation) {
+        this._instrumentations |= instrumentation;
+    }
+
+    public removeInstrumentation(instrumentation: Constants.StatsBeatInstrumentation) {
+        this._instrumentations &= ~instrumentation;
+    }
+
+    public countRequest(duration: number, success: boolean) {
+        if (!this.isEnabled()) {
+            return;
+        }
+        this._totalRequestCount++;
+        this._intervalRequestExecutionTime += duration;
+        if (success === false) {
+            this._totalFailedRequestCount++;
+        }
+        else {
+            this._totalSuccesfulRequestCount++;
+        }
+    }
+
+    public countException() {
+        if (!this.isEnabled()) {
+            return;
+        }
+        this._exceptionCount++;
+    }
+
+    public countThrottle() {
+        if (!this.isEnabled()) {
+            return;
+        }
+        this._throttleCount++;
+    }
+
+    public countRetry() {
+        if (!this.isEnabled()) {
+            return;
+        }
+        this._retryCount++;
+    }
+
+    public trackStatsBeatMetrics() {
+        this._trackRequestDuration();
+        this._trackRequestsCount();
+    }
+
+    private _trackRequestDuration() {
+        var lastRequests = this._lastRequests;
+        var requests = {
+            totalRequestCount: this._totalRequestCount,
+            time: +new Date
+        };
+        var intervalRequests = (requests.totalRequestCount - lastRequests.totalRequestCount) || 0;
+        var elapsedMs = requests.time - lastRequests.time;
+        var averageRequestExecutionTime = ((this._intervalRequestExecutionTime - this._lastIntervalRequestExecutionTime) / intervalRequests) || 0; // default to 0 in case no requests in this interval
```

Comment:
not sure i fully understand this..  can you do this: 
totalRequestDuration/totalRequestcount for each 15 mins?

Reply:
you might need to track total request duration.

Reply:
this is a carbon copy of how we calculate average duration for customer requests for Live Metrics, it is exactly what you mentioned, totalRequestDuration/totalRequestcount for each 15 mins